xmtp/src/builder.rs

```diff
@@ -91,7 +101,13 @@ where
         }
     }
 
-    pub fn build(&mut self) -> Result<Client<P>, ClientBuilderError<P::Error>> {
+    pub fn build(&mut self) -> Result<Client<A, P>, ClientBuilderError<P::Error>> {
+        let api_client =
+            self.api_client
+                .take()
```

Comment:
Taking the value of the Option leaves `self` in a changed state -- and multiple calls will produce different configurations of Clients. `build` should consume self if its going to consume it's members, to avoid confusion